cmd/admin-tier-add.go

```diff
@@ -0,0 +1,332 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cmd
+
+import (
+	"fmt"
+	"io/ioutil"
+
+	"github.com/fatih/color"
+	"github.com/minio/cli"
+	json "github.com/minio/mc/pkg/colorjson"
+	"github.com/minio/mc/pkg/probe"
+	"github.com/minio/minio/pkg/console"
+	"github.com/minio/minio/pkg/madmin"
+)
+
+var adminTierAddFlags = []cli.Flag{
+	cli.StringFlag{
+		Name:  "endpoint",
+		Value: "",
+		Usage: "cloud backend endpoint. e.g https://s3.amazonaws.com",
+	},
+	cli.StringFlag{
+		Name:  "region",
+		Value: "",
+		Usage: "cloud backend region. e.g us-west-2",
+	},
+	cli.StringFlag{
+		Name:  "access-key",
+		Value: "",
+		Usage: "AWS s3 or compatible object storage access-key",
+	},
+	cli.StringFlag{
+		Name:  "secret-key",
+		Value: "",
+		Usage: "AWS s3 or compatible object storage secret-key",
+	},
+	cli.StringFlag{
+		Name:  "account-name",
+		Value: "",
+		Usage: "Azure Blob Storage account name",
+	},
+	cli.StringFlag{
+		Name:  "account-key",
+		Value: "",
+		Usage: "Azure Blob Storage account key",
+	},
+	cli.StringFlag{
+		Name:  "credentials-file",
+		Value: "",
+		Usage: "gcs credentials file used to access/transition objects to GCS",
+	},
+	cli.StringFlag{
+		Name:  "bucket",
+		Value: "",
+		Usage: "cloud backend target bucket",
+	},
+	cli.StringFlag{
+		Name:  "prefix",
+		Value: "",
+		Usage: "cloud backend target prefix",
+	},
+	cli.StringFlag{
+		Name:  "storage-class",
+		Value: "",
+		Usage: "AWS S3 or compatible object storage's storage-class. e.g AWS s3 suppports S3_STANDARD and S3_IA",
+	},
+}
+
+var adminTierAddCmd = cli.Command{
+	Name:   "add",
+	Usage:  "add a new remote tier target",
+	Action: mainAdminTierAdd,
+	Before: setGlobalsFromContext,
+	Flags:  append(globalFlags, adminTierAddFlags...),
+	CustomHelpTemplate: `NAME:
+  {{.HelpName}} - {{.Usage}}
+
+USAGE:
+  {{.HelpName}} TIER_TYPE TARGET NAME [TIER_FLAGS]
+
+NAME:
+  Name of remote tier target. e.g S3TIER
+
+TIER_TYPE:
+  Cloud storage backend where objects specified by bucket lifecycle configuration can be transitioned to.
+  Supported values are s3, azure and gcs.
+
+TIER_FLAGS:
+  Tier type specific flags.
+
+FLAGS:
+  {{range .VisibleFlags}}{{.}}
+  {{end}}
+EXAMPLES:
+  1. Configure a new remote tier which transitions objects to Azure Blob Storage.
+     {{.Prompt}} {{.HelpName}} azure myminio AZTIER --account-name foobar --account-key foobar123 --region us-east-1 --bucket testbucket --prefix testprefix/
+
+  2. Configure a new remote tier which transitions objects to AWS S3 Standard-IA storage class.
```

Comment:
@krisis Need to change the `--storage-class` value here. Also probably referencing `REDUCED_REDUNDANCY` unless we're sure we can support `STANDARD_IA`